tutorials/llm/nemo2-peft.ipynb

```diff
@@ -0,0 +1,2052 @@
+    "\n",
+    "We will be using NeMo-Run to run our experiments in this notebook.\n",
+    "\n",
+    "\n",
+    "# NeMo Tools and Resources\n",
+    "1. [NeMo Github repo](https://github.com/NVIDIA/NeMo)\n",
+    "\n",
+    "2. [NeMo-Run Github repo](https://github.com/NVIDIA/NeMo-Run/)\n",
+    "\n",
+    "3. NeMo Framework Training container: `nvcr.io/nvidia/nemo:dev`\n",
+    "\n",
+    "\n",
+    "\n",
+    "# Educational Resources\n",
+    "1. Blog: [Mastering LLM Techniques: Customization](https://developer.nvidia.com/blog/selecting-large-language-model-customization-techniques/)\n",
+    "\n",
+    "2. Whitepaper: [LoRA: Low-Rank Adaptation of Large Language Models](https://arxiv.org/abs/2106.09685)\n",
+    "\n",
+    "3. [NeMo 2.0 Overview](https://docs.nvidia.com/nemo-framework/user-guide/latest/nemo-2.0/index.html)\n",
+    "\n",
+    "4. Blog: [Tune and Deploy LoRA LLMs with NVIDIA TensorRT-LLM](https://developer.nvidia.com/blog/tune-and-deploy-lora-llms-with-nvidia-tensorrt-llm/)\n",
+    "\n",
+    "\n",
+    "## Software Requirements\n",
+    "\n",
+    "1. Use the latest [NeMo Framework Training container](https://catalog.ngc.nvidia.com/orgs/nvidia/containers/nemo/tags) . Note that you must be logged in to the container registry to view this page.\n",
+    "\n",
+    "2. This notebook uses the container: `nvcr.io/nvidia/nemo:dev`.\n",
+    "\n",
+    "\n",
+    "## Hardware Requirements\n",
+    "Llama3 8B: minimum 1xA100 80G\n",
+    "\n",
+    "\n",
+    "## Data\n",
+    "This notebook uses the SQUAD dataset. For more details about the data refer to [SQuAD: 100,000+ Questions for Machine Comprehension of Text](https://arxiv.org/abs/1606.05250)\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Step 0: Go inside docker container\n",
+    "\n",
+    "You can start and enter the dev container by:\n",
+    "```\n",
+    "docker run --gpus device=1 --shm-size=2g --net=host --ulimit memlock=-1 --rm -it -v ${PWD}:/workspace -w /workspace -v ${PWD}/results:/results nvcr.io/nvidia/nemo:dev bash\n",
+    "\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "\n",
+    "# Step 1: Import HuggingFace checkpoint\n",
+    "First request download permission from Meta and Hugging Face. Log in through `huggingface-cli` using your Hugging Face token before importing llama3 models. \n",
+    "\n",
+    "```\n",
+    "$ huggingface-cli login\n",
+    "```\n",
+    "\n",
+    "Once logged in, you can use the following script to import a Hugging Face model. Based on the provided model configuration (`Llama3-8b` in the example below), the `llm.import_ckpt` API will download the specified model using the \"hf://<huggingface_model_id>\" URL format. It will then convert the model into NeMo 2.0 format. \n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [
```

Comment:
Please clear all notebook outputs unless there is a specific point you want to make with a cell's output.

Reply:
Done, thanks!